mdoc.Test.FSharp/Delegates.fs

```diff
@@ -0,0 +1,119 @@
+﻿module Delegates
+
+// The following code shows the syntax for creating delegates that represent various methods in a class.
+// Depending on whether the method is a static method or an instance method, 
+// and whether it has arguments in the tuple form or the curried form, 
+// the syntax for declaring and assigning the delegate is a little different.
+
+type Test1() =
+  static member add(a : int, b : int) =
+     a + b
+  static member add2 (a : int) (b : int) =
+     a + b
+
+  member x.Add(a : int, b : int) =
+     a + b
+  member x.Add2 (a : int) (b : int) =
+     a + b
+
+let replicate n c = String.replicate n (string c)
+
+
+// Delegate specifications must not be curried types. 
+// Use 'typ * ... * typ -> typ' for multi-argument delegates, 
+// and 'typ -> (typ -> typ)' for delegates returning function values.	
+
+type Delegate1 = delegate of (int * int) -> int// Delegate1 works with tuple arguments.
+type Delegate2 = delegate of int * int -> int // Delegate2 works with curried arguments.
+type Delegate3 = delegate of int * char -> string
+type Delegate4 = delegate of int -> (int -> char)
+type Delegate5 = delegate of int -> (int -> char -> string)
+type Delegate6 = delegate of (int -> float) -> char
+type Delegate7 = delegate of (int -> char -> string) -> float
```

Comment:
I find it interesting that the generated signature for this type is

```fsharp
type Delegates.Delegate7 = delegate of (int -> char -> string) -> double
```

AFAIK, F# supports both `double` and `float` ... so it's curious that the type reported here is `System.Double` for the return type. Can you look into what might be happening here?

Reply:
@joelmartinez yes, you're right, F# supports `double` and `float`, but both of them are `System.Double` in IL code ([type match table](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/primitive-types)). In this case, I've decided to generate `double` cause it seems to be closer to the real IL code. There is a similar issue with `float32` and `single` which really are `System.Single`, I generate `single` for them.